packages/grafana-runtime/src/services/LocationService.tsx

```diff
@@ -162,3 +163,21 @@ export const navigationLogger = navigationLog.logger;
 
 // For debugging purposes the location service is attached to global _debug variable
 attachDebugger('location', locationService, navigationLog);
+
+// Simple context so the location service can be used without being a singleton
+const LocationServiceContext = React.createContext<LocationService | undefined>(undefined);
+
+export function useLocationService(): LocationService {
+  const service = useContext(LocationServiceContext);
+  if (!service) {
+    throw new Error('useLocationService must be used within a LocationServiceProvider');
+  }
+  return service;
+}
+
+export const LocationServiceProvider: React.FC<{ service: LocationService; children: React.ReactNode }> = ({
+  service,
+  children,
+}) => {
+  return <LocationServiceContext.Provider value={service}>{children}</LocationServiceContext.Provider>;
+};
```

Comment:
It would be nice to add a basic test case for the hook and the provider in `LocationService.test.ts(x)`.

Reply:
Yup added a test there